edge-modules/api-proxy-module/src/monitors/config_monitor.rs

```diff
@@ -61,16 +61,25 @@ pub fn start(
                         return Ok(());
                     }
                     Either::Right((Some(Ok(message)), _)) => {
-                        if let azure_iot_mqtt::module::Message::TwinPatch(twin) = message {
-                            if let Err(err) = save_raw_config(&twin) {
-                                error!("received message {}", err);
-                                false
-                            } else {
-                                info!("New config received from twin, reloading the config");
-                                true
-                            }
-                        } else {
-                            false
+                        match message {
+                            azure_iot_mqtt::module::Message::TwinPatch(twin) => {
+                                if let Err(err) = save_raw_config(&twin.properties) {
+                                    error!("received message {}", err);
+                                    false
+                                } else {
+                                    info!("New config received from twin, reloading the config");
+                                    true
+                                }
+                            },
+                            azure_iot_mqtt::module::Message::TwinInitial(twin) => {
+                                if let Err(_) = save_raw_config(&twin.desired.properties) {
+                                    false
```

Comment:
That's a good point, I didn't log this error here on purpose. Because you get initial twin all the time and initial twin may not contain a proxy configuration.